### DISCUSSION TITLE
Adding component to an entity without registry ?

### DISCUSSION THREAD

**Discussion by rudiHammad** [skypjack/entt#1030] [General]
Hello,

I just started using Entt today, this a very basic question but I was wondering it adding components to entities are only done through the registry. For instance, can we do something like this (adding directly the component to an entity)
`like playerEntity.emplace<TransformComponent>(std::vec2(), ...);`
 instead of  
`registry.emplace<TransformComponent>(playerEntity,` ax::Vec2(10, 20));`

I am asking because I recently finished an ECS course were we learned how to implement our own ECS, and we had a registry class as well that did exactly what Entt does, and then we wrapped the Entity class to be able to directly emplace the component.

thx

**Comment by zaucy** [comment]
I believe a [handle](https://github.com/skypjack/entt/blob/cb974bf5671c757cbce2e0248787bad616b06bf9/docs/md/entity.md#handle) is what you're looking for
